OpenRA.Mods.Common/Widgets/Logic/TabCompletionLogic.cs

```diff
@@ -19,58 +19,83 @@ public class TabCompletionLogic
 		IList<string> candidates = new List<string>();
 		int currentCandidateIndex = 0;
 		string lastCompleted;
+		public readonly string PrefixOnStart;
+		public readonly string SuffixOnStart;
 		string prefix;
 		string suffix;
 
-		public IList<string> Commands { get; set; }
+		private int longestElementLength;
+		private IList<string> universe;
+		public IList<string> Universe
+		{
+			get
+			{
+				return universe;
+			}
+
+			set
+			{
+				universe = value;
+				longestElementLength = universe.Aggregate("", (max, cur) => max.Length > cur.Length ? max : cur).Length;
+			}
+		}
 
-		public IList<string> Names { get; set; }
+		public TabCompletionLogic(string prefixOnStart = "", string suffixOnStart = ": ")
```

Comment:
Most clients I've used either append ": " or ", " so one of those two is what I vote for.